### ISSUE TITLE
sig-testing: support auto-tag and auto-bump for test images

### ISSUE THREAD

**Issue by chewong** [kubernetes/kubernetes#100273]
I forget how this works but technically we can't use 2.30 until it's been promoted. Usually there's a followup PR that updates test/images/manifest once promotion has occurred. Is that the more appropriate PR to make this change?

The two-step image promotion dance and fixed/hardcoded version approach used by e2e test images could really stand to be revisited. Not suggesting we try to do it in v1.21 just more thinking out loud. Auto-tagging with date+gitsha is an approach that's worked well for other images in this project AFAIK.

_Originally posted by @spiffxp in https://github.com/kubernetes/kubernetes/pull/99860#discussion_r594784152_

**Comment by BenTheElder** [comment]
I think auto tag is a good idea. I don't think we're ready for auto-promote yet which would block automatically using new tags.

**Comment by BenTheElder** [comment]
This should probably be brought up somewhere with more visibility to see if anyone has concerns with date-sha, but otherwise I think this is
/triage accepted

**Comment by BenTheElder** [comment]
raised in #sig-testing slack to see if anyone else has input

**Comment by spiffxp** [comment]
/milestone v1.23
I think this worth considering now that we've consolidated most images to staging repos

**Comment by NilimaC04** [comment]
Hi 👋 I'm the bug triage shadow for release 1.23.
There hasn't been any activity on this issue for a couple of months now. Just want to make sure this issue is still on track for the 1.23?

**Comment by BenTheElder** [comment]
I don't think this is happening in 1.23 at least. I'm not aware of anyone taking this up.
/milestone clear

**Comment by NilimaC04** [comment]
Thanks for the update @BenTheElder! Can we plan to do this in v1.24?

**Comment by BenTheElder** [comment]
I don't know of anyone planning to work on this currently, so I don't think we can plan on any particular milestone.
